cudax/include/cuda/experimental/__detail/config.cuh

```diff
@@ -63,6 +63,21 @@
 #  define _CUDAX_CONSTEVAL consteval
 #endif
 
+// private nested class types cannot be used as tparams to __global__ functions.
+// _CUDAX_SEMI_PRIVATE expands to public when __CUDACC__ is defined and
+// private otherwise.
+#if defined(__CUDACC__)
+#  define _CUDAX_SEMI_PRIVATE public
+#else // ^^^ __CUDACC__ ^^^ / vvv !__CUDACC__ vvv
+#  define _CUDAX_SEMI_PRIVATE private
+#endif
+
+#if _CCCL_CUDA_COMPILER(CLANG)
+#  define _CUDAX_HOST_DEVICE_DEDUCTION_GUIDE _CCCL_HOST_DEVICE
+#else // ^^^ _CCCL_CUDA_COMPILER(CLANG) ^^^ / vvv !_CCCL_CUDA_COMPILER(CLANG) vvv
+#  define _CUDAX_HOST_DEVICE_DEDUCTION_GUIDE
+#endif // !_CCCL_CUDA_COMPILER(CLANG)
```

Comment:
Lets just drop this and unconditionally mark this as host device. They are technically correct, that a type *could* have separate host / device deduction guides